modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/js/ckeditor5/BalloonEditor.tsx

```diff
@@ -0,0 +1,52 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {BalloonEditor as BaseBalloonEditor, EventInfo} from 'ckeditor5';
+import React from 'react';
+
+import '../../css/ckeditor5/editor.scss';
+
+import {CKEditor} from '@ckeditor/ckeditor5-react';
+
+import getDefaultEditorConfig from './utils/getDefaultEditorConfig';
+import {
+	EEditorConfigPreset,
+	EEditorType,
+	LiferayEditorConfig,
+} from './utils/types';
+
+const BalloonEditor = ({
+	className,
+	config,
+	data,
+	onChange,
+	onReady,
+}: {
+	className?: string;
+	config?: LiferayEditorConfig;
+	data?: string;
+	onChange?: (event: EventInfo, editor: BaseBalloonEditor) => void;
+	onReady?: (editor: BaseBalloonEditor) => void;
+}) => {
+	return (
+		<div className={`lfr-ck ${className ? className : ''}`}>
+			<CKEditor
```

Comment:
My initial idea was to introduce `BaseEditor` in the same way as we have for CKE5. But:
1. It lead to odd quirks with TS, where we need to keep track of both possible versions passed in by inner component. So, I needed to do things like `editor = editor as BalloonEditor;`
2. Once I extracted config logic into `getDefaultEditorConfig`, there was no meaningful logic to have in the scope of `BaseEditor`. CX logic would fit in that place, but it could probably also handled in a util. If it's inconvenient, we can always add inheritance with it.